### DISCUSSION TITLE
chezmoi init - https for pull and ssh for push

### DISCUSSION THREAD

**Discussion by returntrip** [twpayne/chezmoi#3658] [Q&A] [answered]
Is there a way while running `chezmoi init $USER` to setup the remote to https for pulls and to ssh for pushes?

**Comment by orzomaxx** [comment]
This is controlled by `git`, not by `chezmoi`. So you need to use `.gitconfig` to do this

**Comment by twpayne** [comment, marked as answer]
I use [these lines](https://github.com/twpayne/dotfiles/blob/62716405d69c4e286eea891f7b09e720d2938939/home/private_dot_config/git/config.tmpl#L104-L105) in `~/.config/git/config`, which is managed by chezmoi and so populated when you run `chezmoi init`:

```
[url "git@github.com:twpayne/"]
	insteadOf = https://github.com/twpayne/
```
